catalog/core/catalog-core-definitionparser/src/main/java/ddf/catalog/definition/impl/DefinitionParser.java

```diff
@@ -632,12 +640,42 @@ private ValidatorWrapper getValidator(String key, Outer.Validator validator) {
           wrapper.reportingMetacardValidator(relationshipValidator);
           break;
         }
+      case "custom":
+        {
+          List<Outer.Validator> collection = ((Outer.ValidatorCollection) validator).validators;
+          collection.forEach(
+              item -> {
+                AttributeValidator av =
+                    (AttributeValidator)
+                        this.getService(
+                            AttributeValidator.class.getName(),
+                            (String.format("(id=%s)", item.validator)));
+                if (av != null) {
+                  wrapper.attributeValidator(av);
+                }
+              });
+          break;
+        }
       default:
         throw new IllegalStateException("Validator does not exist. (" + validator.validator + ")");
     }
     return wrapper;
   }
 
+  private Object getService(String clazz, String filter) {
+    BundleContext bundleContext = getBundleContext();
+    ServiceReference<?>[] ref;
+    try {
+      ref = bundleContext.getServiceReferences(clazz, filter);
+      if (ref != null && ref.length == 1) {
```

Comment:
As it is currently if there is more than one service reference it will fail to enter this if statement, return null, and then silently not add the attribute validator. 

I would recommend changing this to `ref.length > 0`, Unless you are specifically trying to prevent the case that multiple versions might be registered. While we usually only have one bundle providing a service its always possible there could be more than one, so we should handle that case. In many places we pick the first service reference in the list (which is probably fine) , but you could also do a LOGGER.warn that there were multiple services and we chose the first in the list. (maybe printing the list of service references that we chose from would provide some useful information if they have implemented toString methods)